x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/delete_engine_modal.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React from 'react';
+
+import { useActions, useValues } from 'kea';
+
+import { EuiConfirmModal } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { EnginesListLogic } from './engines_list_logic';
+
+export const DeleteEngineModal: React.FC = () => {
+  const { closeDeleteEngineModal, deleteEngine } = useActions(EnginesListLogic);
+  const {
+    deleteModalEngineName: engineName,
+    isDeleteModalVisible,
+    isDeleteLoading,
+  } = useValues(EnginesListLogic);
+
+  if (isDeleteModalVisible) {
+    return (
+      <EuiConfirmModal
+        title={i18n.translate('xpack.enterpriseSearch.content.engineList.deleteEngineModal.title', {
+          defaultMessage: 'Permanently delete this engine?',
+        })}
+        onCancel={() => {
+          closeDeleteEngineModal();
+        }}
+        onConfirm={() => {
+          deleteEngine({ engineName });
+        }}
+        cancelButtonText="No, cancel"
+        confirmButtonText={i18n.translate(
+          'xpack.enterpriseSearch.content.engineList.deleteEngineModal.confirmButton.title',
+          {
+            defaultMessage: 'Yes, Delete this engine',
```

Comment:
nitpick: 'Delete' should not be capitalized

Reply:
Probably might need to change this to `delete` icon. Mentioned in my previous [comment ](https://github.com/elastic/kibana/pull/148771#discussion_r1068265296)

Reply:
@saarikabhasi Yes, if I am missing a delete icon then that was an oversight on my behalf (lots of screens!) so feel free to add it if that is a consistent pattern across the platform.

Reply:
I guess, from what I have seen so far, consistent pattern would be using  `'Yes, delete this engine'`,  not the `delete` icon

If that okay, will continue with current change that is `'Yes, delete this engine' `

Reply:
@saarikabhasi For transparency, once i make some updates I'll be engaging the UI copy team here: https://github.com/elastic/enterprise-search-team/issues/3702 for a verdict